scale/diagnostic/fixtures/diagnostic_recipe_types.json

```diff
@@ -1,46 +0,0 @@
-[
-   {
-        "model": "recipe.RecipeType",
-        "pk": null,
-        "fields": {
-            "name": "scale-casino",
-            "title": "Scale Casino",
-            "description": "Chains together multiple Scale Roulette jobs. This recipe type can be useful for testing Scale's handling of recipes in the presence of job failures.",
-            "is_system": true,
-            "is_active": true,
-            "definition": {
-                 "version": "6",
-                 "input_data": [],
-                 "nodes": {"roulette-1": {"dependencies":[], "input":{},
-                                          "node_type": {"node_type": "job", "job_type_name": "scale-roulette", "job_type_version": "1.0.0", "job_type_revision": 1}},
-                           "roulette-2": {"dependencies":[{"name": "roulette-1"}], "input":{},
-                                          "node_type": {"node_type": "job", "job_type_name": "scale-roulette", "job_type_version": "1.0.0", "job_type_revision": 1}},
-                           "roulette-3": {"dependencies":[{"name": "roulette-2"}], "input":{},
-                                          "node_type": {"node_type": "job", "job_type_name": "scale-roulette", "job_type_version": "1.0.0", "job_type_revision": 1}}}
-            },
-            "revision_num": 1,
-            "created": "2017-11-13T00:00:00.0Z",
-            "last_modified": "2017-11-13T00:00:00.0Z"
-        }
-    },
-    {
-        "model": "recipe.RecipeTypeRevision",
-        "pk": null,
```

Comment:
Could the problem be the revision?  The model code creates the recipe type and it's first revision and then attempts to create the revision again with this fixture?

Reply:
Problem was with the RecipeType model not defining a `natural_key` method that returned a tuple with just the recipe_type.name value `return (self.name,)` 😑